src/big_uint/chip.rs

```diff
@@ -0,0 +1,644 @@
+use halo2_base::{
+    gates::{GateInstructions, RangeChip, RangeInstructions},
+    halo2_proofs::{circuit::Value, plonk::Error},
+    utils::{biguint_to_fe, decompose_biguint, fe_to_biguint, BigPrimeField},
+    AssignedValue, Context, QuantumCell,
+};
+use halo2_ecc::bigint::{big_is_zero, mul_no_carry, OverflowInteger};
+use num_bigint::BigUint;
+
+use super::{AssignedBigUint, Fresh, Muled, RefreshAux};
+
+#[derive(Clone, Debug)]
+pub struct BigUintChip<'a, F: BigPrimeField> {
+    pub range: &'a RangeChip<F>,
+    pub limb_bits: usize,
```

Comment:
This can be used in other external projects we do as well. So either we refactor this right now to be in a monorepo or keep it as is and we can reuse the library in other projects too.